src/lsstseries/ensemble.py

```diff
@@ -603,37 +677,64 @@ def from_parquet(
             self._err_col = err_col
         if band_col is not None:
             self._band_col = band_col
+        if provenance_col is not None:
+            self._provenance_col = provenance_col
 
         if additional_cols:
             columns = None  # None will prompt read_parquet to read in all cols
+
         else:
             columns = [self._time_col, self._flux_col, self._err_col, self._band_col]
+            if self._provenance_col is not None:
+                columns.append(self._provenance_col)
 
         # Read in the source parquet file(s)
         self._source = dd.read_parquet(
             source_file, index=self._id_col, columns=columns, split_row_groups=True
         )
 
-        if npartitions and npartitions > 1:
-            self._source = self._source.repartition(npartitions=npartitions)
-        elif partition_size:
-            self._source = self._source.repartition(partition_size=partition_size)
-
         if object_file:  # read from parquet files
-            # Read in the object file
-            file_table = dd.read_parquet(object_file, index=self._id_col, split_row_groups=True)
+            # Read in the object file(s)
+            self._object = dd.read_parquet(object_file, index=self._id_col, split_row_groups=True)
+
+            # Handle nobs_band columns
+            if nobs_cols is not None:
+                self._nobs_bands = nobs_cols
+            else:
+                # sets empty nobs cols in object
+                unq_filters = np.unique(self._source[self._band_col])
+                self._nobs_bands = [f"nobs_{filt}" for filt in unq_filters]
+                for col in self._nobs_bands:
+                    self._object[col] = np.nan
```

Comment:
It's not clear to me why you are setting all the values to nan. Why not just leave out the column?

Reply:
The intention here is to define the columns that will house the nobs_{band} information up front. If a user has set the sync_tables flag, then the sync will automatically update these. But if not, then they would still have access to the columns that *will* house that information once a sync has been performed. I'm also thinking this might clear up some of the ambiguity that would come up if a user loads an object table with nobs cols, but doesn't assign them to lsstseries nobs columns. Particularly by showing in the non-sync cases that new columns have been added that will dynamically update, and those aren't the same as the columns with pre-populated values found from the file.

Reply:
Makes sense. Thanks.